File: Algorithms/Algorithms-Problems/BinarySearch/Bishu and Soilders/BishuAndSoilders.java
import java.util.*;
class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		int[] arr = new int[n];
		for (int i = 0; i < n; i++)
			arr[i] = sc.nextInt();
		int k=sc.nextInt();
		Arrays.sort(arr);
		while(k-->0){
			int x = sc.nextInt();
			int sum = 0;
			int count = 0;
			for(int i=0;i<arr.length;i++) {
				if (arr[i]<=x) {
					sum+=arr[i];
					count++;
				}
                else
					break;
			}
			System.out.println(count+" "+sum);
		}
	}
}

File: Algorithms/Algorithms-Problems/BinarySearch/Bishu and Soilders/BishuAndSoilders.cpp
#include <bits/stdc++.h>
using namespace std;

void TakeInput(int*, int);
void CountInput(int*, int);

int main() {
    int ArrSize;

    cout << "\n_________________________________";
    cout << "\nPlease Enter total elements: ";
    cin >> ArrSize;
    cout << "---------------------------------\n";

    int Arr[ArrSize];
    TakeInput(Arr, ArrSize);
    sort(Arr, Arr + ArrSize);
    CountInput(Arr, ArrSize);

    return 0;
}

void TakeInput(int Arr[], int Size) {
    for (int i = 0; i < Size; i++) {
        cin >> Arr[i];
    }
}

void CountInput(int Arr[], int Size) {
    int q, p;

    cin >> q;
    for (int i = 0;i < Size;i++) {
        cin >> p;
        int j = 0, count = 0, sum = 0;

        while(Arr[j] <= p && j < Size) {
            count++;
            sum += Arr[j];
            j++;
        }
        cout << count << " " << sum << endl;
    }
}